staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -273,8 +273,7 @@ func (s *SecureServingOptions) MaybeDefaultWithSelfSignedCerts(publicAddress str
 	}
 	if !canReadCertAndKey {
 		// add either the bind address or localhost to the valid alternates
-		bindIP := s.BindAddress.String()
-		if bindIP == "0.0.0.0" {
+		if s.BindAddress.IsUnspecified() {
 			alternateDNS = append(alternateDNS, "localhost")
```

Comment:
Should this be setting the alternate DNS to [::1], if IPv6 mode, instead of localhost?

Reply:
I'm not an expert regarding this but seems it's always possible to use localhost in place of the loopback IP.